pyroom.1
.TH PyRoom 1 "01 Mar 2009"
.SH NAME
\fBPyRoom\fP \- distraction-free writing

.SH DESCRIPTION
.fi
.B PyRoom
lets you edit text files simply and efficiently in a full-screen window, with no distractions.
.nf
.LP
.SH OPTIONS

.TP
\fB\-\-help\fR
Prints the command line options and exits..
.TP
\fB\-\-version\fR
Prints the programm's version and exits.
.TP
\fBfilename(s)...\fR
Specifies the file to open
.SH BUGS
If you find a bug, please report it at 

.B http://launchpad.net/pyroom
.SH AUTHORS
The PyRoom Team


CHANGELOG
pyroom trunk ()
===============

User visible changes
--------------------

* create nonexistant files if necessary (LP:#307257)
* recently used files are now remembered (LP:#315873)
* saving of custom themes (LP:#321057)
* autosave now saves filename.txt~ instead of to /tmp (LP:#323621)
* removed line numbering (dropped gtksourceview, see LP:#297155)
* offer to restore from autosave backups (LP:#330260)
* per-theme fonts replaced with global font selection
  also offers to use gnome default fonts if applicable, i.e. if gconf
  bindings for gconf are available (LP:#303431)
* allow indenting of paragraph beginnings (LP:#331405)

Noteable background changes
---------------------------

* new method of theme lookup and application (LP:#326108)
* rewrote autosave mechanism (LP:#330045)
* dropped dependency on gtksourceview (LP:#297155)
* translation system should work now (LP:#335418 and LP:#272095)

Bugfixes
--------
* disable background pixmaps that some gtk themes use (LP:#309773)
* actually use name field in theme files instead of filenames (LP:#323617)
* avoid deprecation warning when starting pyroom (LP:#330043)
* stop autosaving when told to (LP:#329888)
* stop fading working incorrectly with dark on light themes (LP:#323613)
* removed outdated information from README (LP:#334958)
* consistent capitalisation of themes (LP:#334004)
* made RecentManager use optional (LP:#336814)

pyroom 0.3.2 (2008-12-03)
=========================
    
User visible changes
--------------------

* added multi-monitor support (LP:#189037)
* textbox background can be configured independantly from app background
  (LP:#285116)
* textbox border can be switched on/off (LP:#193855)

Noteable background changes
---------------------------

* pyroom adheres to XDG directory specifications for config etc
  (LP:#262697)
* standard themes reside in a global dir now, not in homedirs (LP:#246133)
* switched from gtksourceview 1 to 2, removing a lot of gnome 
  dependencies (LP:#300684)

Bugfixes
--------

* don't ask to save empty files (LP:#262798)
* textbox size configurable in percentages, not fractures (LP:#296935)
* can't select textbox sizes >100% any more (LP:#263046)
* switching buffer scrolls to cursor position (LP:#270350)
* cursor placed at file bottom on opened files (LP:#188908)
* allow using pyroom for opening files from GUIS (LP:#282009)


pyroom.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# -----------------------------------------------------------------------------
# PyRoom - A clone of WriteRoom
# Copyright (c) 2007 Nicolas P. Rougier & NoWhereMan
# Copyright (c) 2008 The Pyroom Team - See AUTHORS file for more information
#
# This program is free software: you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation, either version 3 of the License, or (at your option) any later
# version.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
# details.
#
# You should have received a copy of the GNU General Public License along with
# this program.  If not, see <http://www.gnu.org/licenses/>.
# -----------------------------------------------------------------------------

"""
PyRoom - A clone of WriteRoom
=============================

Based on code posted on ubuntu forums by NoWhereMan (www.nowhereland.it)
(Ubuntu thread was "WriteRoom/Darkroom/?")

:copyright: 2007 Nicolas P. Rougier & NoWhereMan Copyright
:copyright: 2008 The PyRoom Theme - See AUTHORS file for more information
:license: GNU General Public License, version 3 or later
"""

__VERSION__ = '0.2'

import gettext
gettext.install('pyroom', 'locale')
import locale
locale.setlocale(locale.LC_ALL, '')
from optparse import OptionParser
import sys

import gtk

import autosave
from basic_edit import BasicEdit
from pyroom_error import handle_error
from preferences import PyroomConfig

pyroom_config = PyroomConfig()

if __name__ == '__main__':
    sys.excepthook = handle_error

    files = []
    style = pyroom_config.config.get('visual', 'theme')
    autosave.autosave_time = pyroom_config.config.get('editor', 'autosavetime')

    # Preparing the themes list for the optionparser
    themes_list = pyroom_config.read_themes_list()
    themes_list.append('custom')

    # Get commandline args
    parser = OptionParser(usage = _('%prog [-v] [--style={style name}] \
[file1] [file2]...'),
                        version = '%prog ' + __VERSION__,
                        description = _('PyRoom lets you edit text files \
simply and efficiently in a full-screen window, with no distractions.'))
    parser.set_defaults(
                        style = pyroom_config.config.get('visual', 'theme'),
                        autosave_time = pyroom_config.config.get('editor',
                                                        'autosavetime')
                       )
    parser.add_option('-a', '--autosave',
                    type = 'int', action = 'store', dest = 'autosave_time',
                    help = _('Specify the amount of time, in minutes, to \
                              automatically save your work.'))
    parser.add_option('-s', '--style',
                    action = 'store', dest = 'style',
                    type = 'choice', choices = themes_list,
                    help = _('Override the default style'))
    (options, args) = parser.parse_args()

    style = options.style
    autosave.autosave_time = options.autosave_time
    files = args

    # Create relevant buffers for file and load them
    pyroom = BasicEdit(style=style, pyroom_config=pyroom_config)
    buffnum = 0
    if len(files):
        for filename in files:
            pyroom.open_file_no_chooser(filename)
            buffnum += 1

    pyroom.set_buffer(buffnum)
    pyroom.status.set_text(
        _('Welcome to Pyroom %s, type Control-H for help' % __VERSION__))
    gtk.main()
